search-api/search-actors/src/main/java/org/sunbird/actors/SearchActor.java

```diff
@@ -104,6 +104,16 @@ private SearchDTO getSearchDTO(Request request) throws Exception {
             if (filters.containsKey("relatedBoards"))
                 filters.remove("relatedBoards");
 
+            if (filters.containsKey(SearchConstants.contentType) && null != filters.get(SearchConstants.contentType)) {
+                List<String> contentTypeList = getList(filters.get(SearchConstants.contentType));
+                Object courseTypeFromFilter = filters.get(SearchConstants.courseType);
```

Comment:
If contentTypeList is not empty and contentTypeList has only one value i.e. `Course`, then only we will capture `courseType`

Reply:
I don't see any impact on reading courseType before that condition. if we put this under such condition, it will result into nested control instruction.

Reply:
code refactored as per above comment.